docs/x509/reference.rst

```diff
@@ -2123,11 +2092,8 @@ X.509 Extensions
         recommendation in :rfc:`5280` section 4.2.1.2.
 
         :param public_key: One of
-            :class:`~cryptography.hazmat.primitives.asymmetric.rsa.RSAPublicKey`,
-            :class:`~cryptography.hazmat.primitives.asymmetric.dsa.DSAPublicKey`,
-            :class:`~cryptography.hazmat.primitives.asymmetric.ec.EllipticCurvePublicKey`,
-            :class:`~cryptography.hazmat.primitives.asymmetric.ed25519.Ed25519PublicKey` or
-            :class:`~cryptography.hazmat.primitives.asymmetric.ed448.Ed448PublicKey`.
+            :data:`~cryptography.hazmat.primitives.asymmetric.types.CertificatePublicKeyTypes`.
+            # FIXME is this correct?
```

Comment:
The type in code of `SubjectKeyIdentifier.from_public_key()` uses `CertificatePublicKeyTypes` (allows X25519PublicKey, X448PublicKey), but documentation instead matches `CertificateIssuerPublicKeyTypes`. Not sure which is correct.

Reply:
This should be `CertificateIssuerPublicKeyTypes` because the key in the parent cert has to allow signing.

Reply:
Updated the type hint in code too.